### PR TITLE
Update maven to 3.9.0

### PR DESCRIPTION
**Pull request by NotMyFault**

### CODE REVIEW

**Review by MarkEWaite — APPROVED** [review]
Maven 3.9.0 successfully compiles the 10+ plugins that I maintain with both Java 11 and Java 17

File: Dockerfile [hunk 1]
```diff
@@ -1,8 +1,8 @@
 FROM eclipse-temurin:11-jdk-focal
 
-ARG MAVEN_VERSION=3.8.7
+ARG MAVEN_VERSION=3.9.0
 # https://downloads.apache.org/maven/maven-3/3.8.7/binaries/apache-maven-3.8.7-bin.tar.gz.sha512
```
Comment (MarkEWaite) [inline Dockerfile]: Could this comment be updated or removed? (optional, non-blocking)

Comment (NotMyFault) [inline Dockerfile]:
```suggestion
# https://downloads.apache.org/maven/maven-3/3.9.0/binaries/apache-maven-3.9.0-bin.tar.gz.sha512
```

Comment (NotMyFault) [reply Dockerfile]: Updated in https://github.com/jenkinsci/core-pr-tester/pull/29/commits/1601905dc74c1b353b0ac44dc8d74cda93113fee

**Pull request merged**